rm-datasource/src/main/java/io/seata/rm/datasource/undo/UndoExecutorFactory.java

```diff
@@ -37,18 +37,31 @@
      * @return the undo executor
      */
     public static AbstractUndoExecutor getUndoExecutor(String dbType, SQLUndoLog sqlUndoLog) {
-        if (!dbType.equals(JdbcConstants.MYSQL)) {
+        if (!dbType.equalsIgnoreCase(JdbcConstants.MYSQL)&&!dbType.equalsIgnoreCase(JdbcConstants.ORACLE)) {
             throw new NotSupportYetException(dbType);
         }
-        switch (sqlUndoLog.getSqlType()) {
-            case INSERT:
-                return new MySQLUndoInsertExecutor(sqlUndoLog);
-            case UPDATE:
-                return new MySQLUndoUpdateExecutor(sqlUndoLog);
-            case DELETE:
-                return new MySQLUndoDeleteExecutor(sqlUndoLog);
-            default:
-                throw new ShouldNeverHappenException();
-        }
+          if(dbType.equalsIgnoreCase(JdbcConstants.ORACLE)) {
+            switch (sqlUndoLog.getSqlType()) {
+                case INSERT:
+                    return new OracleUndoInsertExecutor(sqlUndoLog);
+                case UPDATE:
+                    return new OracleUndoInsertExecutor(sqlUndoLog);
```

Comment:
This doesn't look right. I would guess that the `OracleUndoUpdateExecutor` should be used here.
(Same goes for the line below with delete and `OracleUndoDeleteExecutor`).